test/config-luatest/gh_10636_json_schema_test.lua

```diff
@@ -0,0 +1,68 @@
+local schema = require('experimental.config.utils.schema')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_json_schema = function()
+    local s = schema.new('basic', schema.record({
+        foo = schema.record({
+            bar = schema.enum({
+                '0',
+                '1',
+            }),
+            baz = schema.scalar({
+                type='string',
+                default='0',
+                allowed_values = {'0', '1', '2'},
```

Comment:
Nit: `=` is surrounded by spaces in some places, but it doesn't in some others -- no consistency. Usually we surround it: `{foo = 'bar'}`.

Reply:
Done.